app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentTest.kt

```diff
@@ -368,11 +374,68 @@ class StateFragmentTest {
 
   @Test
   fun testContentCard_forDemoExploration_withCustomOppiaTags_displaysParsedHtml() {
-    ActivityScenario.launch(ContentCardTestActivity::class.java).use {
+    launch(ContentCardTestActivity::class.java).use {
       val htmlResult =
         "Hi, welcome to Oppia! is a tool that helps you create interactive learning activities that can be continually improved over time.\n\n" +
             "Incidentally, do you know where the name 'Oppia' comes from?"
-      onView(atPositionOnView(R.id.state_recycler_view, 0,R.id.content_text_view)).check(matches(hasDescendant(withText(htmlResult))))
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.content_text_view)).check(matches(withText(htmlResult)))
+    }
+  }
+
+  @Test
+  fun testMultipleChoiceInput_showsRadioButtons_forDemoExploration_withCustomOppiaTags_userSelectsDesiredOption() {
+    launch(ContentCardTestActivity::class.java).use {
+      onView(withId(R.id.play_exploration_button_1)).perform(click())
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        actionOnItemAtPosition<InteractionAdapter.MultipleChoiceViewHolder>(0, click())
+      )
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        actionOnItemAtPosition<InteractionAdapter.MultipleChoiceViewHolder>(1, click())
+      )
+    }
+  }
+
+  @Test
+  fun testItemSelectionInput_showsCheckBox_forDemoExploration_withCustomOppiaTags_userSelectsDesiredOptions() {
+    launch(ContentCardTestActivity::class.java).use {
+      onView(withId(R.id.play_exploration_button_2)).perform(click())
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        actionOnItemAtPosition<InteractionAdapter.ItemSelectionViewHolder>(0, click())
+      )
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        actionOnItemAtPosition<InteractionAdapter.ItemSelectionViewHolder>(2, click())
+      )
+    }
+  }
+
+  @Test
+  fun testItemSelectionInput_showsCheckBox_withMaxSelectionAllowed_userSelectsDesiredOptions_correctError() {
+    launch(ContentCardTestActivity::class.java).use {
+      onView(withId(R.id.play_exploration_button_2)).perform(click())
+      onView(withId(R.id.selection_interaction_recyclerview)).perform(
+        actionOnItemAtPosition<InteractionAdapter.ItemSelectionViewHolder>(0, click())
+      )
+      it.onActivity { activity ->
+
+        activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
+        activity.requestedOrientation = Configuration.ORIENTATION_PORTRAIT
```

Comment:
What purpose do these orientation changes serve? If we want to test a single orientation change, we should just do one rather than 2 here.

Reply:
okay it was added while debugging. Removed it.